### ISSUE TITLE
ETH for DAI Swap leads to extremely large PnL Taxable value

### ISSUE THREAD

**Issue by Michael-A-Heuer** [rotki/rotki#5406]
## Problem Definition

The data below is copied from the PnL Report generated for a time window, in which a trade has happened.

| Type | Notes | Location | Timestamp | Asset | Free Amount | Taxable Amount | Price | PnL Taxable | PnL Free | Cost Basis Taxable | Cost Basis Free |
|------|-------|----------|-----------|-------|-------------|----------------|-------|------------|----------|--------------------|------------------|
| transaction event | https://etherscan.io/tx/0x4925c933b493c1782c7f63ed4c3fdf74c45b65bc89d9c1c7bde86b553f17207a  ->  Burned 0.00283269940446163 ETH for gas | blockchain | 05/03/2022 11:43:56 | ETH | 0.00283269940446163 | 0 | 2425.67 | -6.87119396442044 | 6.74628833928054 |  | 0.12490562513990200000000000 |
| transaction event | https://etherscan.io/tx/0x4925c933b493c1782c7f63ed4c3fdf74c45b65bc89d9c1c7bde86b553f17207a  ->  Swap 265.484484606843433397 DAI in uniswap-v3 from 0xEFA21088F3830cD7d9f526eA268f7AabA34F785F | blockchain | 05/03/2022 11:43:56 | eip155:1/erc20:0x6B175474E89094C44Da98b954EedeAC495271d0F | 0E+00 | 265.484484606843000000 | 3.99691419761188E+56 | 106111870577076000000000000000000000000000000000000000000000 | 0 | 227.507785595233 |
| transaction event | https://etherscan.io/tx/0x4925c933b493c1782c7f63ed4c3fdf74c45b65bc89d9c1c7bde86b553f17207a  ->  Approve 1.157920892373161954235709850E+59 DAI of 0xEFA21088F3830cD7d9f526eA268f7AabA34F785F for spending by 0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45 | blockchain | 05/03/2022 11:43:56 | eip155:1/erc20:0x6B175474E89094C44Da98b954EedeAC495271d0F | 1.15792089237316E+59 | 0 | 2.10109328967273E-57 | 0 | 0 |  |

![Bildschirmfoto 2023-01-15 um 16 14 22](https://user-images.githubusercontent.com/20623991/212549825-7209758a-df03-48be-892b-7b5c67ab1fee.png)

This trade was automatically imported but the EUR price with `3.99691419761188E+56` is incorrect.
What would be the best way of fixing this problem?

## Logs
```
[15/01/2023 16:41:09 CET] DEBUG rotkehlchen.api.server Greenlet-4: start rotki api POST /api/1/reports/15/data view_args={'report_id': 15}, query_string=b''
[15/01/2023 16:41:10 CET] DEBUG rotkehlchen.api.server Greenlet-4: end rotki api POST /api/1/reports/15/data view_args={'report_id': 15}, query_string=b'', status_code=200, result={'result': {'entries': [{'type': 'transaction event', 'notes': 'transaction 0x4925c933b493c1782c7f63ed4c3fdf74c45b65bc89d9c1c7bde86b553f17207a Burned 0.00283269940446163 ETH for gas', 'location': 'blockchain', 'timestamp': 1646477036, 'asset': 'ETH', 'free_amount': '0.00283269940446163', 'taxable_amount': '0', 'price': '2425.67', 'pnl_taxable': '-6.8711939644204420421', 'pnl_free': '6.746288339280540119004746100', 'cost_basis': {'is_complete': True, 'matched_acquisitions': [{'amount': '0.00283269940446163', 'event': {'timestamp': 1490361098, 'full_amount': '49.7145000000', 'rate': '44.09420390429351597622424041', 'index': 327}, 'taxable': False}]}}, {'type': 'transaction event', 'notes': 'transaction 0x4925c933b493c1782c7f63ed4c3fdf74c45b65bc89d9c1c7bde86b553f17207a Swap 265.484484606843433397 DAI in uniswap-v3 from 0xEFA21088F3830cD7d9f526eA268f7AabA34F785F', 'location': 'blockchain', 'timestamp': 1646477036, 'asset': 'eip155:1/erc20:0x6B175474E89094C44Da98b954EedeAC495271d0F', 'free_amount': '0E-18', 'taxable_amount': '265.484484606843433397', 'price': '3.996914197611881831066235866E+56', 'pnl_taxable': '1.061118705770765614861604507E+59', 'pnl_free': '0', 'cost_basis': {'is_complete': True, 'matched_acquisitions': [{'amount': '91.278809443468153474', 'event': {'timestamp': 1628838746, 'full_amount': '499.4164124500', 'rate': '0.8558036727373075341949547497', 'index': 5592}, 'taxable': True}, {'amount': '80.2171468700', 'event': {'timestamp': 1628838746, 'full_amount': '80.2171468700', 'rate': '0.8558045589835623631424227442', 'index': 5595}, 'taxable': True}, {'amount': '93.988528293375279923', 'event': {'timestamp': 1634888884, 'full_amount': '339.413685291464276947', 'rate': '0.85905', 'index': 6734}, 'taxable': True}]}, 'group_id': '0x4925c933b493c1782c7f63ed4c3fdf74c45b65bc89d9c1c7bde86b553f17207a1518'}, {'type': 'transaction event', 'notes': 'transaction 0x4925c933b493c1782c7f63ed4c3fdf74c45b65bc89d9c1c7bde86b553f17207a Approve 1.157920892373161954235709850E+59 DAI of 0xEFA21088F3830cD7d9f526eA268f7AabA34F785F for spending by 0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45', 'location': 'blockchain', 'timestamp': 1646477036, 'asset': 'eip155:1/erc20:0x6B175474E89094C44Da98b954EedeAC495271d0F', 'free_amount': '1.15792089237316195423570985E+59', 'taxable_amount': '0', 'price': '2.101093289672732794637489472E-57', 'pnl_taxable': '0', 'pnl_free': '0', 'cost_basis': None, 'group_id': '0x4925c933b493c1782c7f63ed4c3fdf74c45b65bc89d9c1c7bde86b553f17207a1518'},
```

### System Description
Operating system: macOS Monterey
Rotki version: 1.26.3

**Comment by nebolax** [comment]
Thanks for reporting! It will be fixed by #5392

**Comment by Michael-A-Heuer** [comment]
Awesome

**Comment by LefterisJP** [comment]
PR is merged and will be included in 1.27.
